src/Sql/AbstractSql.php

```diff
@@ -1,59 +1,53 @@
 <?php
 /**
- * Zend Framework (http://framework.zend.com/)
- *
- * @link      http://github.com/zendframework/zf2 for the canonical source repository
- * @copyright Copyright (c) 2005-2016 Zend Technologies USA Inc. (http://www.zend.com)
- * @license   http://framework.zend.com/license/new-bsd New BSD License
+ * @see       https://github.com/zendframework/zend-db for the canonical source repository
+ * @copyright Copyright (c) 2005-2019 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-db/blob/master/LICENSE.md New BSD License
  */
 
+declare(strict_types=1);
+
 namespace Zend\Db\Sql;
 
 use Zend\Db\Adapter\Driver\DriverInterface;
 use Zend\Db\Adapter\ParameterContainer;
 use Zend\Db\Adapter\Platform\PlatformInterface;
-use Zend\Db\Sql\Platform\PlatformDecoratorInterface;
 use Zend\Db\Adapter\Platform\Sql92 as DefaultAdapterPlatform;
+use Zend\Db\Sql\Exception\InvalidArgumentException;
+use Zend\Db\Sql\Exception\RuntimeException;
+use Zend\Db\Sql\Platform\PlatformDecoratorInterface;
+use function count;
+use function gettype;
+use function implode;
+use function is_array;
+use function is_string;
+use function preg_replace;
+use function rtrim;
+use function sprintf;
+use function vsprintf;
 
 abstract class AbstractSql implements SqlInterface
 {
-    /**
-     * Specifications for Sql String generation
-     *
-     * @var string[]|array[]
-     */
+    /** @var string[]|array[] Specifications for Sql String generation */
     protected $specifications = [];
 
-    /**
-     * @var string
-     */
+    /** @var string[] */
     protected $processInfo = ['paramPrefix' => '', 'subselectCount' => 0];
 
-    /**
-     * @var array
-     */
+    /** @var array */
     protected $instanceParameterIndex = [];
 
-    /**
-     * {@inheritDoc}
-     */
-    public function getSqlString(PlatformInterface $adapterPlatform = null)
+    public function getSqlString(?PlatformInterface $adapterPlatform = null) : string
     {
-        $adapterPlatform = ($adapterPlatform) ?: new DefaultAdapterPlatform;
+        $adapterPlatform = $adapterPlatform ?: new DefaultAdapterPlatform;
         return $this->buildSqlString($adapterPlatform);
     }
 
-    /**
-     * @param PlatformInterface $platform
-     * @param null|DriverInterface $driver
-     * @param null|ParameterContainer $parameterContainer
-     * @return string
-     */
     protected function buildSqlString(
-        PlatformInterface $platform,
-        DriverInterface $driver = null,
-        ParameterContainer $parameterContainer = null
-    ) {
+        PlatformInterface  $platform,
```

Comment:
I am not sure about this parameters alignment, but if we are gonna go that way here we are missing on space